samcli/lib/build/app_builder.py

```diff
@@ -682,7 +682,7 @@ def _build_function(  # pylint: disable=R1710
                     # None represents the global build image for all functions/layers
                     global_image = self._build_images.get(None)
                     image = self._build_images.get(function_name, global_image)
-
+                    build_workflow = specified_workflow if supports_specified_workflow(specified_workflow) else None
```

Comment:
Shouldn't we set this higher up at line 650?

Reply:
Then can we re-use `specified_workflow` name or something else? Since `build_workflow` is too generic comparing to where it is coming from.